support-frontend/assets/components/paypalExpressButton/PayPalOneClickCheckoutButton.jsx

```diff
@@ -0,0 +1,139 @@
+  } = addressActionCreatorsFor('billing');
+
+  dispatch(setEmail(payPalUserDetails.email));
+  dispatch(setFirstName(payPalUserDetails.firstName));
+  dispatch(setLastName(payPalUserDetails.lastName));
+  dispatch(setAddressLineOne(payPalUserDetails.shipToStreet));
+  dispatch(setTownCity(payPalUserDetails.shipToCity));
+  dispatch(setState(payPalUserDetails.shipToState));
+  dispatch(setPostcode(payPalUserDetails.shipToZip));
+  // $FlowIgnore stoopid flow
+  dispatch(setCountry(payPalUserDetails.shipToCountryCode));
+};
+
+function mapStateToProps(state: CheckoutState, ownProps) {
+  return {
+    hasLoaded: state.page.checkout.payPalHasLoaded,
+    csrf: state.page.csrf,
+    productPrices: state.page.checkout.productPrices,
+    currencyId: state.common.internationalisation.currencyId,
+    isTestUser: state.page.checkout.isTestUser,
+    amount: finalPrice(
+      state.page.checkout.productPrices,
+      state.common.internationalisation.countryId,
+      ownProps.billingPeriod,
+    ).price,
+    submissionError: state.page.checkout.submissionError,
+    wasClicked: state.page.checkout.billingPeriod === ownProps.billingPeriod,
+  };
+}
+
+function mapDispatchToProps() {
+  return {
+    setupRecurringPayPalPayment: setupSubscriptionPayPalPaymentWithShipping,
+    onPayPalCheckoutCompleted: (payPalCheckoutDetails: PayPalCheckoutDetails) =>
+      (dispatch: Dispatch<Action>, getState: () => CheckoutState) => {
+        if (payPalCheckoutDetails.user) {
+          updateStore(dispatch, payPalCheckoutDetails.user);
+        }
+        onPaymentAuthorised(
+          {
+            paymentMethod: PayPal,
+            token: payPalCheckoutDetails.baid,
+          },
+          dispatch,
+          getState(),
+        );
+      },
+    onClick: (billingPeriod, trackingId, product) => (dispatch: Dispatch<Action>) => {
+      const componentId = `${trackingId}-${billingPeriod}-${product}-PayPal`;
+      trackCheckoutSubmitAttempt(
+        componentId,
+        product,
+        PayPal,
+        product,
+      );
+      return dispatch(formActionCreators.setBillingPeriod(billingPeriod));
+    },
+  };
+}
+
+const payPalButton = css`
+  box-sizing: border-box;
+  margin-top: ${space[3]}px;
+`;
+
+function PayPalOneClickCheckoutButton(props: PropTypes) {
+  const submissionErrorHeading = 'Sorry there was a problem';
+  const errorReason = props.submissionError && 'amazon_pay_fatal';
+  return (
+    <div css={payPalButton}>
+      {props.wasClicked && <GeneralErrorMessage
+        errorReason={errorReason}
+        errorHeading={submissionErrorHeading}
+      />}
+      <PayPalExpressButton
+        onPayPalCheckoutCompleted={props.onPayPalCheckoutCompleted}
+        csrf={props.csrf}
+        currencyId={props.currencyId}
+        hasLoaded={props.hasLoaded}
+        canOpen={() => true}
+        onClick={() => props.onClick(props.billingPeriod, props.trackingId, props.product)}
+        formClassName="form--contribution"
```

Comment:
Yes I don't really know why it's there. I've removed it.

Reply:
Actually I'm going to add it back in because it's a required prop and I even though it doesn't look like it's actually used, I don't want to get into removing it in this PR